src/treeChooser/treeChooser.controller.js

```diff
@@ -135,6 +140,13 @@ function TreeChooserController(
         }
         break;
       case 9: //Tab
+        //if we have matches in list choose first one
+        var presentItems = vm.getPresentItems();
+        if (presentItems.length) {
+          var childs = [];
+          presentItems[0].getAllLeafs(childs);
```

Comment:
Don't assume that we only want leaves. You should just create a function called ```getBestMatch``` that gets the first visible node or leaf (depending on whether only leafs are selectable or not)